homeassistant/components/shelly/entity.py

```diff
@@ -20,6 +20,25 @@ def temperature_unit(block_info: dict) -> str:
     return TEMP_CELSIUS
 
 
+def shelly_naming(self, block, entity_type: str, channels: int):
+    """Naming for switch and sensors."""
+    if channels > 1:
+        entity_name = self.wrapper.device.settings["relays"][int(self.block.channel)][
```

Comment:
Should we update aioshelly to always cast `channel` to an int ?

Reply:
Yup, you have my vote.

Reply:
Please consider that the real device channels start from 1.
To be consistent with what is displayed on the device web and printed on the device we need to add +1 to the channel number.
This can be done in aioshelly or when creating the entity, in my opinion it is better that the aioshelly will not add +1 and only do it when creating  the entity.

Reply:
Agree, will change that.

 I'm not so happy with the code. I feel that it can be cleaned a lot. But at least it works now ;-)

Reply:
Done